### PR TITLE
initial support for bentley cra5

### PR DESCRIPTION
**Pull request by aruniverse**
Update test-apps to use @benltey/react-scripts@5.0.0-dev.x

- Webpack 5 removed node polyfills, in a previous pr I removed usage of most node apis in the core pkgs, looks like core-common still has a few deps on some of those core node modules (stream/buffer/zlib/os etc). we need to pull those out of the common pkgs (most of them are used for the rpc work) into its own pkg 
  - i looked into using the new `node:` scheme to make it more explicit we're using core node modules, but need to modify the webpack config for that. will look into that in a separate pr 
- Previously we were webpacking the frontends of the test-apps to get localization and other assets to show up, i've fixed this in our fork; should just be able to compile with tsc for the frontend, should cut down our ci/build times
- we were never running the ecschema-rpc-interface tests, fixed to be run in the pipeline
- was able to prune all but 1 audit exclusion

### CODE REVIEW

**Comment by saskliutas** [comment]
@aruniverse I ran presentation-test-app and looks like everything works. I just noticed some warnings and errors from webpack when running `npm run start:webserver` but looks like they do not impact app.

```
WARNING in ../../common/temp/node_modules/.pnpm/sax@1.2.4/node_modules/sax/lib/sax.js 
163:13-37
Module not found: Error: Can't resolve 'stream' in 'D:\iModelJs\ImodelJs-2\itwinjs-core\common\temp\node_modules\.pnpm\sax@1.2.4\node_modules\sax\lib'
```

```
WARNING in ../../core/common/lib/cjs/rpc/web/WebAppRpcLogging.js 23:23-36
Module not found: Error: Can't resolve 'os' in 'D:\iModelJs\ImodelJs-2\itwinjs-core\core\common\lib\cjs\rpc\web'
```

```
ERROR in Plugin "react" was conflicted between ".eslintrc.js" and "BaseConfig » D:\iModelJs\ImodelJs-2\itwinjs-core\common\temp\node_modules\.pnpm\eslint-config-react-app@7.0.0_e57ff54e94f85ef9cbfbd7ef3ac06963\node_modules\eslint-config-react-app\base.js". 
```

**Comment by pmconne** [comment]
@DStradley will test display-performance-test-app in ImageTests today.
@hnn0003 will test nightly display-performance-test-runs.

**Comment by DStradley** [comment]
@aruniverse I got these warnings compiling display-performance-test-app:

> Compiled with warnings.
> 
> Module not found: Error: Can't resolve 'stream' in 'D:\iModeljs\ImageTests\imodeljs\common\temp\node_modules\.pnpm\sax@1.2.4\node_modules\sax\lib'
> BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
> This is no longer the case. Verify if you need this module and configure a polyfill for it.
> 
> If you want to include a polyfill, you need to:
> 	- add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
> 	- install 'stream-browserify'
> If you don't want to include a polyfill, you can use an empty module like this:
>   ...1 line omitted...
> 
> Module not found: Error: Can't resolve 'os' in 'D:\iModeljs\ImageTests\imodeljs\core\common\lib\cjs\rpc\web'
> BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
> This is no longer the case. Verify if you need this module and configure a polyfill for it.
> 
> If you want to include a polyfill, you need to:
> 	- add a fallback 'resolve.fallback: { "os": require.resolve("os-browserify/browser") }'
> 	- install 'os-browserify'
> If you don't want to include a polyfill, you can use an empty module like this:
> 	resolve.fallback: { "os": false }

It seems to be running though.  I'll check the results of the ImageTests run tomorrow.

**Comment by DStradley** [comment]
@aruniverse display-test-app compiles without messages, but does not run (at least in electron anyway).
When using a dev server, I see the same warning messages in the server console as I got compiling display-performance-test-app.
But dev server or no, it just comes up and hangs with a blank white window (and title bar).

**Comment by hnn0003** [comment]
> @hnn0003 will test nightly display-performance-test-runs.
@pmconne & @aruniverse & @DStradley 
So, when initially I set up an auto dpta test to run, I got this error (see [here](https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1373855&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b&t=516e2b04-73ee-5606-2739-4144c3e63ae7)):
image

After talking with David and seeing the error he got when compiling, I tried changing the version of node that the suto test was using to 12.22.x instead of the 12.18.3 that it was using. In doing this, the error appears to have changed to be perhaps the same as what David's seeing, where it's just a warning now, although the auto dpta test appears to treat this warning as a failure anyways, stopping the test from proceeding (see [here](https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1373889&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b&t=516e2b04-73ee-5606-2739-4144c3e63ae7):
image

**Comment by aruniverse** [comment]
Thank you @saskliutas , @DStradley , @hnn0003 , @bsteinbk for testing it out

Re:
> ERROR in Plugin "react" was conflicted between ".eslintrc.js" and "BaseConfig 

This can be ignored for now, I need to go through and update the eslint version to match the same major ver used by cra

> WARNING in ../../common/temp/node_modules/.pnpm/sax@1.2.4/node_modules/sax/lib/sax.js 
163:13-37
Module not found: Error: Can't resolve 'stream' in 'D:\iModelJs\ImodelJs-2\itwinjs-core\common\temp\node_modules\.pnpm\sax@1.2.4\node_modules\sax\lib'

This comes from a sub dep on `xml-js`, looks like its being used by @mdastous-bentley for `WmtsCapabilities`. Michel can you test this out and make sure this functions properly?

> WARNING in ../../core/common/lib/cjs/rpc/web/WebAppRpcLogging.js 23:23-36
Module not found: Error: Can't resolve 'os' in 'D:\iModelJs\ImodelJs-2\itwinjs-core\core\common\lib\cjs\rpc\web'

This warning can be ignored imo; its just due to how we couple code to run in both node and browser, we handle the error correctly imo

Looks like the pipelines are failing due to using an outdated node ver. @williamkbentley can we bump the docs pipeline to use node 16? 12 is reaching EOL at the end of next month. 

DPTA is using `path` so I added a polyfill for it, ideally thatd be removed. @DStradley or @hnn0003 would one of you be able to removing use of `path`? Do you guys need to webpack the frontend for your auto dpta tests? Otherwise we can switch the build step for that to just run `tsc` like the other apps

**Comment by aruniverse** [comment]
> @aruniverse display-test-app compiles without messages, but does not run (at least in electron anyway). But dev server or no, it just comes up and hangs with a blank white window (and title bar).

@DStradley , after working with @hnn0003 and updating your node ver were you able to run DTA? I'm able to run it fine, and not sure why it'd show up as a blank window for you? Have you tried refreshing it with `f5`?

**Comment by DStradley** [comment]
@aruniverse, no my node was already at 12.22.  BTW, the ImageTests run I did last night (using display-performance-test-app) did run fine, just had the build message.
As to display-test-app though, I use it to capture shader output during some ImageTests builds, and that did not run on that machine (hung and got killed after timeout).  I still cannot run it either on my machine.  I built with build:ci, and I normally run it with a production server, but also tried the dev server.  F5 did not help, and bringing up the debug window would not let me break into it.

**Comment by markschlosseratbentley** [comment]
@aruniverse @DStradley I built the branch on Windows and cannot run display-test-app using Node 12.17.  I did the following and it hangs at "Starting the development server..."
```
rush purge
rush install
rush build --to display-test-app
cd test-apps/display-test-app
npm run start
```
If I install Node 16, it runs and I can open an iModel.  Loads of error in the server terminal, though:
```
D:\itwinjs-core\test-apps\display-test-app>npm run start

> display-test-app@1.8.0 start
> cross-env NODE_ENV=development run-p start:webserver start:electron


> display-test-app@1.8.0 start:webserver
> cross-env DISABLE_NEW_JSX_TRANSFORM=true SKIP_PREFLIGHT_CHECK=true  BROWSER=none USE_FULL_SOURCEMAP=true DISABLE_ESLINT=true TRANSPILE_DEPS=false USE_FAST_SASS=true DISABLE_TERSER=true react-scripts start


> display-test-app@1.8.0 start:electron
> electron ./lib/backend/DtaElectronMain.js


(node:12432) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
(Use `electron --trace-warnings ...` to show where the warning was created)
(node:20916) [DEP_WEBPACK_DEV_SERVER_ON_AFTER_SETUP_MIDDLEWARE] DeprecationWarning: 'onAfterSetupMiddleware' option is deprecated. Please use the 'setupMiddlewares' option.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:20916) [DEP_WEBPACK_DEV_SERVER_ON_BEFORE_SETUP_MIDDLEWARE] DeprecationWarning: 'onBeforeSetupMiddleware' option is deprecated. Please use the 'setupMiddlewares' option.
Starting the development server...
Compiled with warnings.

Module not found: Error: Can't resolve 'stream' in 'D:\itwinjs-core\common\temp\node_modules\.pnpm\sax@1.2.4\node_modules\sax\lib'
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }

Module not found: Error: Can't resolve 'os' in 'D:\itwinjs-core\core\common\lib\cjs\rpc\web'
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "os": require.resolve("os-browserify/browser") }'
        - install 'os-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "os": false }

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

assets by path sprites/*.png 182 KiB 14 assets
assets by path cursors/*.cur 10 KiB
  asset cursors/look.cur 2.19 KiB [emitted] [from: ../../core/frontend/lib/public/cursors/look.cur] [copied]
  + 7 assets
assets by path locales/en/*.json 50.4 KiB
  asset locales/en/CoreTools.json 16 KiB [emitted] [from: ../../core/frontend/lib/public/locales/en/CoreTools.json] [copied]
  + 6 assets
assets by path assets/ 120 KiB 6 assets
assets by path images/*.svg 9.39 KiB
  asset images/cesium-ion.svg 4.59 KiB [emitted] [from: ../../core/frontend/lib/public/images/cesium-ion.svg] [copied]
  + 3 assets
assets by path *.svg 2.68 KiB
  asset section-marker.svg 963 bytes [emitted] [from: ../../core/hypermodeling/lib/public/section-marker.svg] [copied]
  + 3 assets
+ 5 assets
orphan modules 38.2 KiB [orphan] 27 modules
runtime modules 28.2 KiB 13 modules
modules by path ../../core/ 11.5 MiB 944 modules
modules by path ../../common/temp/node_modules/.pnpm/ 1.31 MiB 276 modules
modules by path ../../presentation/common/lib/cjs/ 255 KiB 73 modules
modules by path ./ 424 KiB 58 modules
modules by path ../../ui/appui-abstract/lib/cjs/ 220 KiB 49 modules
modules by path ../../editor/ 379 KiB
  modules by path ../../editor/frontend/lib/cjs/ 369 KiB 12 modules
  modules by path ../../editor/common/lib/cjs/*.js 9.88 KiB 3 modules
external "electron" 42 bytes [built] [code generated]
semver (ignored) 15 bytes [built] [code generated]
../node/require-utils.node (ignored) 15 bytes [built] [code generated]

WARNING in ../../common/temp/node_modules/.pnpm/sax@1.2.4/node_modules/sax/lib/sax.js 163:13-37
Module not found: Error: Can't resolve 'stream' in 'D:\itwinjs-core\common\temp\node_modules\.pnpm\sax@1.2.4\node_modules\sax\lib'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }
 @ ../../common/temp/node_modules/.pnpm/xml-js@1.6.11/node_modules/xml-js/lib/xml2js.js 1:10-24
 @ ../../common/temp/node_modules/.pnpm/xml-js@1.6.11/node_modules/xml-js/lib/index.js 3:13-32
 @ ../../core/frontend/lib/cjs/tile/map/WmtsCapabilities.js 15:17-34
 @ ../../core/frontend/lib/cjs/tile/internal.js 73:13-46
 @ ../../core/frontend/lib/cjs/core-frontend.js 120:13-39
 @ ./src/frontend/DisplayTestApp.ts 9:0-68 118:49-89 118:90-111 148:42-69
 @ ./src/index.ts 8:0-35

WARNING in ../../core/common/lib/cjs/rpc/web/WebAppRpcLogging.js 23:23-36
Module not found: Error: Can't resolve 'os' in 'D:\itwinjs-core\core\common\lib\cjs\rpc\web'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "os": require.resolve("os-browserify/browser") }'
        - install 'os-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "os": false }
 @ ../../core/common/lib/cjs/rpc/web/WebAppRpcProtocol.js 17:27-56
 @ ../../core/common/lib/cjs/core-common.js 153:13-51
 @ ./src/frontend/DisplayTestApp.ts 8:0-77 58:28-49 76:2-34 78:2-43 116:50-81
 @ ./src/index.ts 8:0-35

2 warnings have detailed information that is not shown.
Use 'stats.errorDetails: true' resp. '--stats-error-details' to show it.

webpack 5.70.0 compiled with 2 warnings in 29265 ms
No issues found.
```

**Comment by pmconne** [comment]
@markschlosseratbentley the [prereqs for iTwin.js](https://github.com/iTwin/itwinjs-core) suggest Node 14 is required. Did you try that?

> an installation of the latest security patch of Node 14. The Node installation also includes the npm package manager.

@calebmshafer do we still (want to) officially support Node 12?

**Comment by calebmshafer** [comment]
> > @calebmshafer do we still (want to) officially support Node 12?
> 
> Node 12 is nearing EOL, it won't be supported past 4/30/22, I don't think we should either.

@aruniverse agreed, I don't want to update our package.json engines field to prevent people from doing so but definitely agree no more fixes should be done purely for Node 12.

**Comment by markschlosseratbentley** [comment]
@pmconne @aruniverse @calebmshafer Just tried Node 14.9.1.  It works similar to Node 16 - builds / runs / opens iModel, but same errors are output to the server terminal (see pasted in earlier comment).
@DStradley

**Comment by aruniverse** [comment]
> @pmconne @aruniverse @calebmshafer Just tried Node 14.19.1. It works similar to Node 16 - builds / runs / opens iModel, but same errors are output to the server terminal (see pasted in earlier comment). @DStradley

They're just warnings, and i've gone over them a bit [here](https://github.com/iTwin/itwinjs-core/pull/3419#issuecomment-1082136605). react-scripts@5 is more verbose by default compared to @4, thats why you're seeing:

```
assets by path sprites/*.png 182 KiB 14 assets
assets by path cursors/*.cur 10 KiB
  asset cursors/look.cur 2.19 KiB [emitted] [from: ../../core/frontend/lib/public/cursors/look.cur] [copied]
  + 7 assets
assets by path locales/en/*.json 50.4 KiB
  asset locales/en/CoreTools.json 16 KiB [emitted] [from: ../../core/frontend/lib/public/locales/en/CoreTools.json] [copied]
  + 6 assets
assets by path assets/ 120 KiB 6 assets
assets by path images/*.svg 9.39 KiB
  asset images/cesium-ion.svg 4.59 KiB [emitted] [from: ../../core/frontend/lib/public/images/cesium-ion.svg] [copied]
  + 3 assets
assets by path *.svg 2.68 KiB
  asset section-marker.svg 963 bytes [emitted] [from: ../../core/hypermodeling/lib/public/section-marker.svg] [copied]
  + 3 assets
+ 5 assets
orphan modules 38.2 KiB [orphan] 27 modules
runtime modules 28.2 KiB 13 modules
modules by path ../../core/ 11.5 MiB 944 modules
modules by path ../../common/temp/node_modules/.pnpm/ 1.31 MiB 276 modules
modules by path ../../presentation/common/lib/cjs/ 255 KiB 73 modules
modules by path ./ 424 KiB 58 modules
modules by path ../../ui/appui-abstract/lib/cjs/ 220 KiB 49 modules
modules by path ../../editor/ 379 KiB
  modules by path ../../editor/frontend/lib/cjs/ 369 KiB 12 modules
  modules by path ../../editor/common/lib/cjs/*.js 9.88 KiB 3 modules
external "electron" 42 bytes [built] [code generated]
semver (ignored) 15 bytes [built] [code generated]
../node/require-utils.node (ignored) 15 bytes [built] [code generated]
```

Some of the other output in your terminal might be coming from the electron process

**Comment by markschlosseratbentley** [comment]
@DStradley in summary, it sounds like you will need Node 14 or greater.

**Comment by pmconne** [comment]
> in summary, it sounds like you will need Node 14 or greater.

So @aruniverse @calebmshafer are we supporting node 12 or not?

**Comment by calebmshafer** [comment]
> > in summary, it sounds like you will need Node 14 or greater.
> 
> So @aruniverse @calebmshafer are we supporting node 12 or not?

@pmconne yes, we're not going to fully drop support (i.e. make it impossible for you to use by updating our engine field) but you will need to use react-scripts@4 in order for it to continue to work.

**Comment by DStradley** [comment]
@aruniverse I installed Node 14.19.1, pulled and rebuilt cra5, and display-test-app still doesn't work on my machine.  Is there some other dependency I might be behind on that would cause this?  (Master still works for me.)

**Comment by hnn0003** [comment]
Hey @aruniverse,
What is the minimum we can do to get DPTA to work with this PR?

Running dpta _locally_ with the changes in this PR works right now, even though it produces warnings when it builds. However, the auto dpta runs (like this one [here](https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1383307&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b&t=516e2b04-73ee-5606-2739-4144c3e63ae7) gets the same warnings seen when you locally run dpta, yet they fail the yaml build script. This is problematic, as we'd obviously like to have our daily auto dpta tests still run successfully after your PR merges with the main branch so we can continue to track data. At the same time, we'd rather not have to rewrite significant portions of dpta right now (as we're committed to other priorities) to have that happen when it was already working before. :P

**Comment by aruniverse** [comment]
> @aruniverse I installed Node 14.19.1, pulled and rebuilt cra5, and display-test-app still doesn't work on my machine. Is there some other dependency I might be behind on that would cause this? (Master still works for me.)

Does it still show up as an empty window for you? I'm surprised even after updating to node 14, DTA doesnt work for you.

**Comment by aruniverse** [comment]
> Running dpta _locally_ with the changes in this PR works right now, even though it produces warnings when it builds. However, the auto dpta runs (like this one [here](https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1383307&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b&t=516e2b04-73ee-5606-2739-4144c3e63ae7) gets the same warnings seen when you locally run dpta, yet they fail the yaml build script. This is problematic, as we'd obviously like to have our daily auto dpta tests still run successfully after your PR merges with the main branch so we can continue to track data. At the same time, we'd rather not have to rewrite significant portions of dpta right now (as we're committed to other priorities) to have that happen when it was already working before. :P

_I think_ we just need to set `"allowWarningsInSuccessfulBuild": true` in the common\config\rush\command-line.json
[common\config\rush\command-line.json](https://github.com/iTwin/itwinjs-core/blob/master/common/config/rush/command-line.json#L44)
image

**Comment by aruniverse** [comment]
> _I think_ we just need to set `"allowWarningsInSuccessfulBuild": true` in the common\config\rush\command-line.json [common\config\rush\command-line.json](https://github.com/iTwin/itwinjs-core/blob/master/common/config/rush/command-line.json?rgh-link-date=2022-04-01T19%3A57%3A28Z#L44) image

@hnn0003 , I added that option for the `build:ci` command which is being used in your pipeline. hopefully it runs your tests. I'll be watching the pipeline run

**Comment by DStradley** [comment]
> Does it still show up as an empty window for you? I'm surprised even after updating to node 14, DTA doesnt work for you.

Yes it does, same symptoms as before.  I am surprised as well.  I even tried re-cloning cra5.

**Comment by hnn0003** [comment]
> @hnn0003 , I added that option for the `build:ci` command which is being used in your pipeline. hopefully it runs your tests. I'll be watching the pipeline run

Awesome, thanks! [Here](https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1383799&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b) is the link to the new run I just kicked off. Hoping it works!!

**Comment by aruniverse** [comment]
> > Does it still show up as an empty window for you? I'm surprised even after updating to node 14, DTA doesnt work for you.
> 
> Yes it does, same symptoms as before. I am surprised as well. I even tried re-cloning cra5.

Hate to ask, but have you tried node 16?

**Comment by markschlosseratbentley** [comment]
> > > Does it still show up as an empty window for you? I'm surprised even after updating to node 14, DTA doesnt work for you.
> > 
> > 
> > Yes it does, same symptoms as before. I am surprised as well. I even tried re-cloning cra5.
> 
> Hate to ask, but have you tried node 16? Is anyone else on the team able to run DTA with cra5?

@DStradley Maybe try 'rush purge' just for a good measure?

**Comment by DStradley** [comment]
> @DStradley Maybe try 'rush purge' just for a good measure?

@markschlosseratbentley, I would think that doing a complete new clone and build to a new directory (which is what I meant by re-clone) should be just as good.

> @aruniverse I can run with Node 14.19.1 if you look at earlier comments from me on this PR.

@aruniverse , yes, which is why I tried that, and also why I am wondering if there is some other dependency that I might need to update besides node?

**Comment by DStradley** [comment]
@aruniverse @markschlosseratbentley, I just tried it fresh on the test machine as well and got the exact same symptoms. (It was upgraded to node 14.19.1 yesterday.)

**Comment by aruniverse** [comment]
> @aruniverse @markschlosseratbentley, I just tried it fresh on the test machine as well and got the exact same symptoms. (It was upgraded to node 14.19.1 yesterday.)

Just to confirm, you copied the `.env` on your new clone / machine right? Shouldn't need any additional deps than whats in the branch + Node 14 & higher

**Comment by DStradley** [comment]
> > @aruniverse @markschlosseratbentley, I just tried it fresh on the test machine as well and got the exact same symptoms. (It was upgraded to node 14.19.1 yesterday.)
> 
> Just to confirm, you copied the `.env` on your new clone / machine right? Shouldn't need any additional deps than whats in the branch + Node 14 & higher

Yes, actually I use a batch script that also clones the imodeljs-config and copies it over during the build.  I also build:ci if that matters.  So we can reproduce this on 2 machines, and not on 2 others.

**Comment by hnn0003** [comment]
Running a new test of the auto dpta with the latest changes here: https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1394365&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b

**Comment by hnn0003** [comment]
> Running a new test of the auto dpta with the latest changes here: https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1394365&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b

So, the test has completed successfully. :) You can go to the link to see the csv files produced if you like, but here's a summary of the results:

 Average Effective FPS of cra5 branch:
--------------------------------------
Chrome: 95.22359658
Electron: 79.13793162
Firefox: 44.2745359


 Average Effective FPS of auto test run from 4/7/22 found [here](https://app.powerbi.com/groups/2fc3ba7e-a05a-4cc0-9411-e9d2d2652ec1/reports/9eef6d45-d0d6-4484-966d-3aff6711b8e4/ReportSection)
--------------------------------------
Chrome: 91
Electron: 77
Firefox: 44

**Comment by DStradley** [comment]
@aruniverse the latest changes you made fixed my display-test-app problem.  (We found the difference on my machines was that I was using npm run start:electron)

**Review by ColinKerr — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]
@aruniverse  I tested the ui in browser and electron and everything seemed to work fine.  I did see the following warnings that I have not seen before.
```Running command:
node D:\github\cra5\itwinjs-core\tools\build\scripts\pseudolocalize --englishDir ./public/locales/en --out ./build/public/locales/en-PSEUDO
Trace | core-backend.IModelHost | IModelHost.startup
(node:11632) [DEP_WEBPACK_DEV_SERVER_ON_AFTER_SETUP_MIDDLEWARE] DeprecationWarning: 'onAfterSetupMiddleware' option is deprecated. Please use the 'setupMiddlewares' option.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:11632) [DEP_WEBPACK_DEV_SERVER_ON_BEFORE_SETUP_MIDDLEWARE] DeprecationWarning: 'onBeforeSetupMiddleware' option is deprecated. Please use the 'setupMiddlewares' option.
Starting the development server...
```

**Comment by aruniverse** [comment]
> @aruniverse I tested the ui in browser and electron and everything seemed to work fine. I did see the following warnings that I have not seen before.

It's coming from sub dep we have, ill add it to my list to investigate and fix before the official cra@5 release

**Comment by hnn0003** [comment]
Currently running the latest code here: https://bentleycs.visualstudio.com/iModelTechnologies/_build/results?buildId=1406583&view=logs&j=1e88f028-fd86-50d4-7db7-032ccdbd7f1b
I'll post the avg Effective FPS once that run finishes and if it's within an expected range of performance. :)

**Comment by DStradley** [comment]
@aruniverse, I tested your latest changes on my machine for display-test-app, and I believe everything is as expected:
- If I do a build, I can npm run start, but npm run start:electron just hangs with an error (since production optimized was not built).
- If I do a build:ci, both npm run start, and npm run start:electron work as expected (debug vs. optimized)

So, I think display-test-app is fine.

**Comment by hnn0003** [comment]
Average expected FPS on latest cra5:

Chrome: 91.6739
Electron: 77.67815641
FireFox: 44.13936923

That looks very similar to the regular runs we've had, so I think it's good from the dpta test perspective. =)

**Review by DStradley — APPROVED** [review]

**Comment by pmconne** [comment]
@aruniverse are you waiting on anything other than outstanding approvals?

**Comment by aruniverse** [comment]
> @aruniverse are you waiting on anything other than outstanding approvals?

nope just need one final approval. ill merge and get it updated.

@wgoehrig @calebmshafer any last concerns?

**Comment by calebmshafer** [comment]
> > @aruniverse are you waiting on anything other than outstanding approvals?
> 
> nope just need one final approval. ill merge and get it updated.
> 
> @wgoehrig @calebmshafer any last concerns?

@aruniverse reviewing now. Mind merging in the meantime?

File: common/config/azure-pipelines/templates/core-build.yaml
Comment (calebmshafer) [inline common/config/azure-pipelines/templates/core-build.yaml:39]: Is this still needed? I see that you added the webpacking to the `build:ci` command.

Comment (aruniverse) [reply common/config/azure-pipelines/templates/core-build.yaml:39]: it's only needed for mac builds because mac doesnt run `build:ci`, i guess i could filter this to be run only on mac, but this only takes 20seconds, and the build should be cached for windows and linux

Comment (calebmshafer) [reply common/config/azure-pipelines/templates/core-build.yaml:39]: gross... I won't block this PR for that but we should re-visit this. I'm still not a fan of build and build:ci

File: test-apps/display-performance-test-app/package.json
Comment (calebmshafer) [inline test-apps/display-performance-test-app/package.json:62]: Are we safe to release the 5.0.0 version of it or are we going to wait on 3.2.0 release first?

File: test-apps/display-test-app/package.json
Comment (calebmshafer) [inline test-apps/display-test-app/package.json:19]: any reason to actually separate these? they were previously both doing a webpack for the frontend... I'll immediately resolve this as per my other comment but we should re-visit this.

**Review by calebmshafer — COMMENTED** [review]
LGTM overall. One comment on timing of a release of the new react-scripts version.

Also hooray (!!) for all the audit ignore being gone

File: test-apps/display-performance-test-app/package.json
Comment (aruniverse) [reply test-apps/display-performance-test-app/package.json:62]: I spoke with Bill G about this. I was thinking we can get an official stable 5.0 release for 3.3/3.4. 

Main people asking for this was Stefan for the showcase and Mike B for electron bundling.

Comment (pmconne) [reply test-apps/display-performance-test-app/package.json:62]: So we are removing this from the 3.2 release plan?

Comment (calebmshafer) [reply test-apps/display-performance-test-app/package.json:62]: I think this PR should still be merged today since it's us testing the 5.0.0 version but it just wouldn't be fully released.

@aruniverse if that's the case, I'll approve this PR and we can get it merged but not mention the support in NextVersion.md yet. We'll save that for 3.3.

**Review by calebmshafer — APPROVED** [review]

**Pull request merged**
